tools/codegen/selective_build/selector.py

```diff
@@ -0,0 +1,154 @@
+from typing import Dict, Set, List, Optional
+import yaml
+
+from dataclasses import dataclass
+
+from tools.codegen.selective_build.operator import *
+
+# A SelectiveBuilder holds information extracted from the selective build
+# YAML specification.
+#
+# It includes information about the build's selectivity, the debug_info
+# associated with this selective build (opaque string), and the set of
+# operators that should be included in the build.
+#
+@dataclass(frozen=True)
+class SelectiveBuilder:
+
+    # If true, then the build is not selective, and includes all
+    # operators.
+    include_all_operators: bool
+
+    # Debug Information at the selective/custom build level.
+    debug_info: Optional[List[str]]
```

Comment:
Is it possible to get more descriptive types here (and in other parts) than just `str`? For example, operator names in the main model are represented as a dataclass, rather than as just strings.

Also, avoid putting mutable types like `List` inside frozen dataclasses, they let you backdoor mutation. Make them tuples instead.

Reply:
> Is it possible to get more descriptive types here (and in other parts) than just `str`? For example, operator names in the main model are represented as a dataclass, rather than as just strings.

I did have something more fine grained in an earlier version of this change, but @ljk53 and I felt like having this be more opaque would be better since this is going to be part of OSS. Thoughts?

> 
> Also, avoid putting mutable types like `List` inside frozen dataclasses, they let you backdoor mutation. Make them tuples instead.

Oh! Thanks - I didn't know that List was mutable - thanks for pointing it out! I'll change it to `Tuple[str...]` or `Sequence[str]`?

Reply:
> I did have something more fine grained in an earlier version of this change, but @ljk53 and I felt like having this be more opaque would be better since this is going to be part of OSS. Thoughts?

There is nothing wrong with being opaque, but then there should be other indicators (e.g., leading underscores) that indicate that the representation is opaque and shouldn't be leaking beyond the file.

> Oh! Thanks - I didn't know that List was mutable - thanks for pointing it out! I'll change it to Tuple[str...] or Sequence[str]?

Tuple is better, as Sequence will let you store lists as the internal rep which means that they won't be hashable.

Reply:
> There is nothing wrong with being opaque, but then there should be other indicators (e.g., leading underscores) that indicate that the representation is opaque and shouldn't be leaking beyond the file.

Got it! I'll use underscores for things of this nature (fixed).

> Tuple is better, as Sequence will let you store lists as the internal rep which means that they won't be hashable.

Thanks! Changed to Tuple.